operator/v1/types_console.go

```diff
@@ -293,11 +293,18 @@ type Perspective struct {
 	// Example: "dev", "admin".
 	// The available perspective ids can be found in the code snippet section next to the yaml editor.
 	// Incorrect or unknown ids will be ignored.
+	// +kubebuilder:validation:XValidation:rule="self.id == 'admin'",message="pinnedResources are not supported for id "admin""
```

Comment:
Can we move this validation rule to the `PinnedResource` or to the type to have more context?

The validation rule should also warn only when `PinnedResource` is defined or has length > 1, right?

```suggestion
	// +kubebuilder:validation:XValidation:rule="self.id == 'admin' && !has(self.pinnedResources) : true",message="pinnedResources are not supported for id "admin""
```

Reply:
Does this work where it is currently, typically we would by adding the validation rule to the `Perspective` struct itself rather than fields within it